server/auth_function/lambda_function.py

```diff
@@ -163,13 +163,32 @@ def populate_user_if_necessary(db_connection, event) -> None:
     else:
         user_name = event["request"]["userAttributes"]["custom:idp_username"]
 
+    cursor = db_connection.cursor()
+
+    # in the case of historical FAM user that has no user_guid stored, add their user_guid
+    # if user does not exist or user has a user_guid already, this update query will do nothing
+    query_add_guid= """
```

Comment:
ideally, if we do a one-time load to add user_guid for all existing users, there is no need for this logic

Reply:
fyi, there are 594 users in FAM PROD db that has no user_guid

Reply:
I agree, that will become a historical record that never get update in the future. But I agree it's a very rare case.